lib/cli/kit/command_registry.rb

```diff
@@ -1,71 +1,145 @@
+# typed: true
 require 'cli/kit'
 
 module CLI
   module Kit
     class CommandRegistry
+      extend(T::Sig)
+
       attr_reader :commands, :aliases
 
+      module ContextualResolverInterface
+        extend(T::Sig)
+        extend(T::Helpers)
+        interface!
+
+        sig { abstract.returns(T::Array[String]) }
+        def command_names; end
+
+        sig { abstract.returns(T::Hash[String, String]) }
+        def aliases; end
+
+        sig { abstract.params(name: String).returns(T.nilable(Class)) }
+        def command_class(name); end
+      end
+
       module NullContextualResolver
+        extend(T::Sig)
+        extend(ContextualResolverInterface)
+
+        sig { implementation.returns(T::Array[String]) }
         def self.command_names
           []
         end
 
+        sig { implementation.returns(T::Hash[String, String]) }
         def self.aliases
           {}
         end
 
+        sig { implementation.params(_name: String).returns(T.nilable(Class)) }
         def self.command_class(_name)
           nil
         end
       end
 
+      sig do
+        params(
+          default: T.nilable(String),
+          contextual_resolver: T.untyped,
+        ).void
+      end
       def initialize(default:, contextual_resolver: nil)
         @commands = {}
```

Comment:
You can also type attributes if you want: https://sorbet.org/docs/type-annotations#declaring-class-and-instance-variables

(it's not mandatory for sigils < strict)

Reply:
Yeah, I think I'm going to but I'm moving up one strictness level at a time.